err_fmt_test.go

```diff
@@ -0,0 +1,863 @@
+package clues_test
+
+import (
+	stderr "errors"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/pkg/errors"
+)
+
+const ()
+
+func plusRE(lines ...string) string {
+	var s string
+
+	for i := 0; i < len(lines); i += 2 {
+		s += lines[i]
+
+		if len(lines[i+1]) > 0 {
+			s += `\t.+/` + lines[i+1]
+		}
+	}
+
+	return s
+}
+
+type checkFmt struct {
+	tmpl     string
+	expect   string
+	reExpect *regexp.Regexp
+}
+
+func (c checkFmt) check(t *testing.T, err error) {
+	t.Run(c.tmpl, func(t *testing.T) {
+		result := fmt.Sprintf(c.tmpl, err)
+
+		if len(c.expect) > 0 && result != c.expect {
+			fmt.Printf("\n-----\n%s\n-----\n%s\n-----", c.expect, result)
+			t.Errorf("unexpected format for template %#v\nexpected \"%s\"\ngot \"%s\"", c.tmpl, c.expect, result)
+		}
+
+		if c.reExpect != nil && !c.reExpect.MatchString(result) {
+
+			fmt.Printf("\n-----\n%s\n-----\n", result)
+			fmt.Printf("\n-----\n\"%v\"\n-----\n%#v\n-----", c.reExpect, result)
+			t.Errorf("unexpected format for template %#v\nexpected \"%v\"\ngot %#v", c.tmpl, c.reExpect, result)
+		}
+	})
+}
+
+var globalSentinel = errors.New("sentinel")
+
+func makeOnion(base error, mid, top func(error) error) error {
```

Comment:
🧅

Reply:
🧅 🔪 🥲

Reply:
that just makes me want to make a `diceOnion()` and integrate it into the test code somehow 🤔